sample/src/main/kotlin/app/futured/donutsample/ui/playground/compose/PlaygroundComposeActivity.kt

```diff
@@ -96,151 +106,160 @@ class PlaygroundComposeActivity : AppCompatActivity() {
-        for (index in 0 until radioGroup.childCount) {
-            if (radioGroup.getChildAt(index).id == radioGroup.checkedRadioButtonId) {
-                val progress = findViewById<SeekBar>(R.id.anim_duration_seekbar).progress
-                updateAnimationBuilders(progress, index)
-                break
-            }
+    private fun updateAnimationSpecs() {
+        val animationDurationMs = findViewById<SeekBar>(R.id.anim_duration_seekbar).progress
+        val layoutAnimSpec = findViewById<RadioGroup>(R.id.layout_anim_spec_radio_group).let {
+            getLayoutAnimationSpec(index = it.checkedButtonIndex(), animDurationMs = animationDurationMs)
+        }
+        val colorAnimSpec = findViewById<RadioGroup>(R.id.color_anim_spec_radio_group).let {
+            getColorAnimationSpec(index = it.checkedButtonIndex(), animDurationMs = animationDurationMs)
+        }
+
+        mutateConfig { donutConfig ->
+            donutConfig
+                .copyWithLayoutAnimationsSpec(layoutAnimSpec)
+                .copyWithColorAnimationsSpec(colorAnimSpec)
         }
     }
 
-    private fun updateAnimationBuilders(progress: Int, index: Int) {
-        val animationBuilders = listOf(
-            PhysicsBuilder<Float>().apply {
-                dampingRatio = Spring.DampingRatioMediumBouncy
-                stiffness = Spring.StiffnessLow
-                displacementThreshold = 0.0001f
-            },
-            TweenBuilder<Float>().apply {
-                easing = FastOutSlowInEasing
-            },
-            TweenBuilder<Float>().apply {
-                easing = LinearOutSlowInEasing
-            },
-            TweenBuilder<Float>().apply {
-                easing = FastOutLinearInEasing
-            },
-            TweenBuilder<Float>().apply {
-                easing = LinearEasing
-            },
-            customAnimationBuilder
+    private fun getLayoutAnimationSpec(index: Int, animDurationMs: Int): AnimationSpec<Float> {
+        val specs = listOf<AnimationSpec<Float>>(
+            snap(),
+            tween(animDurationMs, easing = LinearEasing),
+            spring(Spring.DampingRatioMediumBouncy, Spring.StiffnessLow),
+            tween(animDurationMs, easing = FastOutSlowInEasing),
+            tween(animDurationMs, easing = LinearOutSlowInEasing),
+            tween(animDurationMs, easing = FastOutLinearInEasing)
         )
 
-        val builders = animationBuilders.map { builder ->
-            if (builder is TweenBuilder) {
-                builder.duration = progress
-            }
-            builder
-        }
+        return specs[index]
+    }
+
+    private fun getColorAnimationSpec(index: Int, animDurationMs: Int): AnimationSpec<Color> {
+        val specs = listOf<AnimationSpec<Color>>(
+            snap(),
+            tween(animDurationMs, easing = LinearEasing),
+            spring(Spring.DampingRatioMediumBouncy, Spring.StiffnessLow),
```

Comment:
Spring pre color animácie moc nedáva zmysel.

Reply:
Ide o to že ak už chceme umožniť animovať farby, tak ten consumer si tam už vie vložiť akýkoľvek animation spec. Alebo myslíš iba v kontexte sample appky?

Reply:
jj iba v kontexte sample apky

Reply:
sure, môžem to vyhodiť